src/Avalonia.Native/AvnString.cs

```diff
@@ -61,6 +63,21 @@ private unsafe void EnsureNative()
             }
         }
     }
+
+    internal class AvnStringArray : NativeCallbackBase, IAvnStringArray
+    {
+        private readonly IReadOnlyList<string> _items;
+
+        public AvnStringArray(IReadOnlyList<string> items)
+        {
+            _items = items;
+        }
+
+        public string[] ToStringArray() => (_items as string[] ?? _items.ToArray());
+
+        public uint Count => (uint)_items.Count;
+        public IAvnString Get(uint index) => _items[(int)index].ToAvnString();
```

Comment:
Please check if AvnString instance is actually getting disposed.

Reply:
Fixed with GetNSArrayOfStringsAndRelease + explicit disposal of strings